window.cpp

```diff
@@ -143,14 +144,31 @@ void Window::loadRailway()
 
 void Window::saveRailwayAs()
 {
-	QString fileName = QFileDialog::getSaveFileName(this, tr("Save File"),
-							   "/home/jana/untitled.png",
-							   tr("Railway2 File (*.rly2)"));
+	QString fileName = "";
+	if (drawingArea->getTrackTotal() != 0)
+	{
+		if (drawingArea->checkAllTracksInMapConnected())
+		{
+			fileName = QFileDialog::getSaveFileName(this, tr("Save File"),
+								   "./",
```

Comment:
Suggest not using hard coded paths, for current directory: `QDir::currentPath()`